src/settings.c

```diff
@@ -365,6 +367,62 @@ int read_config(char *config_name, settings_t *settings)
     return 0;
 }
 
+int read_database(char *database_name, settings_t *settings)
+{
+    json_error_t error;
+    const char *section;
+    size_t index;
+    json_t *j_value;
+    json_t *j_entry;
+
+    json_t *database_json = json_array();
```

Comment:
json_array() allocates memory, but the variable gets re-assigned immediately, meaning memory is leaked.